### PR TITLE
remove *test_*.go from vendor

### PR DESCRIPTION
**Pull request by jessfraz**
ping @tianon @calavera 

see https://github.com/docker/libnetwork/pull/484#issuecomment-137169354

### CODE REVIEW

File: hack/.vendor-helpers.sh [hunk 1]
```diff
@@ -118,6 +118,7 @@ clean() {
 
 	echo -n 'pruning unused files, '
 	find vendor -type f -name '*_test.go' -exec rm -v '{}' +
+	find vendor -type f -name '*test_*.go' -exec rm -v '{}' +
```
Comment (thaJeztah) [inline hack/.vendor-helpers.sh]: Just browsing on my phone, so ignore me if it's silly; will the leading `*` also be matching, e.g. `latest_version.go`? Could be a problem if it does :smile:

Comment (jfrazelle) [inline hack/.vendor-helpers.sh]: whoa smart

Comment (thaJeztah) [inline hack/.vendor-helpers.sh]: Hehe, that's why we have reviews :+1:

**Comment by calavera** [comment]
LGTM

**Comment by tianon** [comment]
Does `go build` ignore these files?  I think we discussed this change before and decided against it because `*_test.go` is the only set that `go build` actually ignores, so if someone were to replace these files with the full repo contents they'd actually get a slightly different result, right?

**Comment by jfrazelle** [comment]
yeah but then what do we do about libnetwork

**Comment by tianon** [comment]
Shouldn't that file be renamed upstream to be `utils_test.go` so that it's properly ignored by the compiler except when running tests?

**Comment by tianon** [comment]
or moved into a separate `testutils` package?

**Comment by jfrazelle** [comment]
so it can't be _test.go, but ya maybe `testutils` ping @mrjana @mavenugo

**Comment by tianon** [comment]
Any particular reason why it can't be `_test.go`?

**Comment by mavenugo** [comment]
@tianon because the utils varies based on the os/architecture & we need the test_windows.go, test_freebsd.go, etc....

Including these files under `testutils` package in libnetwork is doable. Should we still have to implement pruning this package while vendoring in ?

**Comment by tianon** [comment]
If it's moved to a separate package and only imported from tests, it should
be automatically excluded by the "go list" logic. :+1:

**Comment by mavenugo** [comment]
@tianon ah. got it. we can do that.

**Pull request closed without merging**